### ISSUE TITLE
Investigate the memory leak and increase the storage space

### ISSUE THREAD

**Issue by hyeongjun-dev** [DeSpread/snapshot-helper#1]
## Description
Files around 10GB~100GB works fine, but if user try on a large file like this over more than 100GB, suspect memory leak

**Comment by hyeongjun-dev** [comment]
I'm tracking this issue under branch
https://github.com/DeSpread/snapshot-helper/tree/bugfix/1-fix-memory-leak

Docker image name with tag: `snapshot-helper:develop-v0.0.7`

**Comment by hyeongjun-dev** [comment]
### Simple Test
I tested on a bare-metal machine with under-spec
- spec : https://www.hetzner.com/dedicated-rootserver/ax102

Scenario:
1. dummy file: 100GB
2. dummy file: 200GB
3. dummy file: 300GB

![스크린샷 2024-10-29 오후 5 53 33](https://github.com/user-attachments/assets/4ad70ba2-5a02-47a1-9cc5-2b3a3144f416)

![스크린샷 2024-10-29 오후 5 53 40](https://github.com/user-attachments/assets/9f08ca9a-db93-4be9-9c81-1b4d3d6dc16f)

All tests passed

**Comment by hyeongjun-dev** [comment]
In testing with `develop-v0.0.7`, Aptos snapshot data of about 600 GB was successfully compressed and uploaded to S3. I checked that the same checksum with md5sum on both the compressed data from Polkachu and uploaded data with its own compressed.

I've also found that performance drops more rapidly for files larger than 10GB than when running in the Unix shell.
As a result, I've concluded that it's possible to compress and upload all 10GB+ files to S3, but I do not recommend this way.
I've added this important note to the README file.

![image](https://github.com/user-attachments/assets/a5cffa9a-34a6-42fe-b137-b29b7ef64ab9)
